tests/config_test.go

```diff
@@ -495,17 +500,20 @@ var _ = Describe("[Serial][rfe_id:899][crit:medium][vendor:cnv-qe@redhat.com][le
 				Expect(err).ToNot(HaveOccurred())
 				defer expecter.Close()
 
-				res, err := expecter.ExpectBatch([]expect.Batcher{
+				res, err := tests.ExpectBatchWithValidatedSend(expecter, []expect.Batcher{
 					// mount iso Secret image
 					&expect.BSnd{S: "sudo su -\n"},
-					&expect.BExp{R: "\\#"},
+					&expect.BExp{R: tests.PromptExpression},
 					&expect.BSnd{S: "mount /dev/sda /mnt\n"},
+					&expect.BExp{R: tests.PromptExpression},
 					&expect.BSnd{S: "echo $?\n"},
 					&expect.BExp{R: tests.RetValue("0")},
-					&expect.BSnd{S: "grep \"PRIVATE KEY\" /mnt/ssh-privatekey\n"},
+					&expect.BSnd{S: "grep -c \"PRIVATE KEY\" /mnt/ssh-privatekey\n"},
+					&expect.BExp{R: tests.RetValue(`[1-9]\d*`)},
 					&expect.BSnd{S: "echo $?\n"},
 					&expect.BExp{R: tests.RetValue("0")},
-					&expect.BSnd{S: "grep ssh-rsa /mnt/ssh-publickey\n"},
+					&expect.BSnd{S: "grep -c ssh-rsa /mnt/ssh-publickey\n"},
+					&expect.BExp{R: tests.RetValue(`[1-9]\d*`)},
```

Comment:
Since the next line makes sure the `grep` succeeded (`echo $?`). I find the `grep -c` and count check confusing. I would choose one of the three options -
1. Using regular grep and waiting for prompt.
2. Using the grep -c and removing the `echo $?`.
3. 
```
&expect.BSnd{S: "cat /mnt/ssh-publickey\n"},
&expect.BExp{"ssh-rsa"},
```
and removing the `echo $?" part.